WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -146,20 +154,25 @@ final class ShippingLabelFormViewModel {
             return
         }
         updateRowState(type: .packageDetails, dataState: .validated, displayMode: .editable)
+
+        // We reset the carrier and rates selected because if the package change
+        // the carrier and rate change accordingly
+        handleCarrierAndRatesValueChanges(selectedRate: nil, selectedSignatureRate: nil, selectedAdultSignatureRate: nil, editable: false)
     }
 
     func handleCarrierAndRatesValueChanges(selectedRate: ShippingLabelCarrierRate?,
                                            selectedSignatureRate: ShippingLabelCarrierRate?,
-                                           selectedAdultSignatureRate: ShippingLabelCarrierRate?) {
+                                           selectedAdultSignatureRate: ShippingLabelCarrierRate?,
+                                           editable: Bool) {
         self.selectedRate = selectedRate
         self.selectedSignatureRate = selectedSignatureRate
         self.selectedAdultSignatureRate = selectedAdultSignatureRate
 
-        guard selectedRate != nil else {
-            updateRowState(type: .shippingCarrierAndRates, dataState: .pending, displayMode: .editable)
+        guard selectedRate != nil || selectedSignatureRate != nil || selectedAdultSignatureRate != nil else {
```

Comment:
I had to think about this one for a bit, but I'm not sure there is any better way to golden path "one of these three rates must be !nil"